### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by AinaArd**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -7,4 +7,5 @@ server.port=8080
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 
-token.expired=86400
\ No newline at end of file
+token.expired=86400
+auth.header=AUTH
```
Comment (Dimanaux) [inline src/main/resources/application.properties]: It's typically called 'Authorization'

File: src/main/java/ru/itis/services/WishListService.java [hunk 2]
```diff
@@ -21,13 +21,16 @@ public WishListService(WishListRepository wishListRepository, UserService userSe
...
+        if (userService.findUserByToken(token).isPresent()) {
+            User currentUser = userService.findUserByToken(token).get();
+            WishList newWL = WishList.builder()
+                    .title(title)
+                    .author(currentUser)
+                    .build();
+            currentUser.getWishLists().add(newWL);
+            return wishListRepository.save(newWL);
+        }
+        return null;
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Bad, bad Aina! Never use null

File: src/main/java/ru/itis/services/WishListService.java [hunk 3]
```diff
@@ -37,10 +40,12 @@ public WishList addNewWishList(String title, String token) {
...
-            currentUser.getWishLists().remove(wishListCandidate.get());
-            userService.save(currentUser);
-            wishListRepository.delete(wishListCandidate.get());
+            if (userService.findUserByToken(token).isPresent()) {
+                User currentUser = userService.findUserByToken(token).get();
+                currentUser.getWishLists().remove(wishListCandidate.get());
+                userService.save(currentUser);
+                wishListRepository.delete(wishListCandidate.get());
+            }
             return true;
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Read about [command and query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation)

File: src/main/java/ru/itis/services/WishListService.java [hunk 4]
```diff
@@ -21,13 +21,16 @@ public WishListService(WishListRepository wishListRepository, UserService userSe
...
-        User currentUser = userService.findUserByToken(token).get();
-        WishList newWL = WishList.builder()
-                .title(title)
-                .author(currentUser)
-                .build();
-        currentUser.getWishLists().add(newWL);
-        return wishListRepository.save(newWL);
+        if (userService.findUserByToken(token).isPresent()) {
+            User currentUser = userService.findUserByToken(token).get();
+            WishList newWL = WishList.builder()
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: `LocalManagerEntityContainerFactoryBean`? Why not just `new WishList(title, currentUser)`?

File: src/main/java/ru/itis/services/WishListService.java [hunk 5]
```diff
@@ -21,13 +21,16 @@ public WishListService(WishListRepository wishListRepository, UserService userSe
...
 
     public WishList addNewWishList(String title, String token) {
-        User currentUser = userService.findUserByToken(token).get();
-        WishList newWL = WishList.builder()
-                .title(title)
-                .author(currentUser)
-                .build();
-        currentUser.getWishLists().add(newWL);
-        return wishListRepository.save(newWL);
+        if (userService.findUserByToken(token).isPresent()) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: It's too procedural.

File: src/main/java/ru/itis/services/UserService.java [hunk 6]
```diff
@@ -35,18 +34,18 @@ public UserService(UsersRepository usersRepository, PasswordEncoder passwordEnco
...
+                .login(userDto.getLogin())
                 .password(hashPassword)
                 .role(Role.AUTHOR)
                 .build();
         usersRepository.save(newUser);
     }
 
-    public TokenDto login(UserForm userForm) {
-        Optional<User> userCandidate = usersRepository.findByLogin(userForm.getLogin());
+    public TokenDto login(UserDto userDto) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/UserService.java]: What does exactly this procedure do?

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java [hunk 7]
```diff
@@ -8,6 +8,7 @@
 import java.io.IOException;
 
 public class TokenAuthenticationFilter implements Filter {
+
```
Comment (Dimanaux) [inline src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: Drop this empty line.

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java [hunk 8]
```diff
@@ -8,6 +8,7 @@
...
     private final static String AUTH_HEADER = "AUTH";
```
Comment (Dimanaux) [inline src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: It's an error. You have this "AUTH" in properties.

File: src/main/java/ru/itis/security/SecurityConfig.java [hunk 9]
```diff
@@ -22,15 +20,14 @@
 @ComponentScan("ru.itis") // говорим, чтобы искал все компоненты в наших пакетах
```
Comment (Dimanaux) [inline src/main/java/ru/itis/security/SecurityConfig.java]: Useless comments.

File: src/main/java/ru/itis/dto/ResponseUserDto.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+@Data
+@Builder
+public class ResponseUserDto {
+
+    private String login;
+    private String password;
+    private List<WishList> wishLists;
+
+    public static ResponseUserDto from(User user) {
+        return ResponseUserDto.builder()
```
Comment (Dimanaux) [inline src/main/java/ru/itis/dto/ResponseUserDto.java]: why not just `new Response(user, wishLists)`?

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 11]
```diff
@@ -35,8 +35,8 @@ public WishList getWishList(@PathVariable("list-id") Long listId, @RequestHeader
     @PostMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Add new item to wish list")
-    public void addNewItem(@PathVariable("list-id") Long listId, ItemForm itemForm, @RequestHeader(name = "AUTH") String token) {
-        itemService.addNewItem(itemForm, listId);
+    public void addNewItem(@PathVariable("list-id") Long listId, ItemDto itemDto, @RequestHeader(name = "AUTH") String token) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: You have "AUTH" as a variable and as a property, why hardcoding it?

File: src/main/java/ru/itis/configuration/Config.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package ru.itis.configuration;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class Config {
+
```
Comment (Dimanaux) [inline src/main/java/ru/itis/configuration/Config.java]: Drop this line.

File: src/main/java/ru/itis/security/SecurityConfig.java (hunk 9 above)
Comment (AinaArd) [reply src/main/java/ru/itis/security/SecurityConfig.java]: fixed

File: src/main/java/ru/itis/configuration/Config.java (hunk 12 above)
Comment (AinaArd) [reply src/main/java/ru/itis/configuration/Config.java]: fixed

**Comment by Dimanaux** [comment]
Also your interfaces very inconsistent. In some cases you want just title and then you create wish list, in some cases you want user record (but not just username and password).

**Comment by Dimanaux** [comment]
Also you PR is too large, programmers are to lazy to read large PRs and you project will end up with bad code in master branch.

File: src/main/java/ru/itis/security/SecurityConfig.java [hunk 13]
```diff
@@ -10,50 +10,40 @@
...
 import ru.itis.security.providers.TokenAuthenticationProvider;
 
-@Configuration // говорим, что у нас конфигурационный класс
-@EnableGlobalMethodSecurity(prePostEnabled = true) // включаем проверку безопасности через аннотации
-@EnableWebSecurity // включаем безопасность
-@ComponentScan("ru.itis") // говорим, чтобы искал все компоненты в наших пакетах
+@Configuration
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@EnableWebSecurity
+@ComponentScan("ru.itis")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/security/SecurityConfig.java]: maybe ru.itis.kpfu?

File: src/main/java/ru/itis/dto/ResponseUserDto.java (hunk 10 above)
Comment (AinaArd) [reply src/main/java/ru/itis/dto/ResponseUserDto.java]: gonna fix later by mapstrust

File: src/main/resources/application.properties (hunk 1 above)
Comment (AinaArd) [reply src/main/resources/application.properties]: fixed

File: src/main/java/ru/itis/services/WishListService.java (hunk 2 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/services/WishListService.java (hunk 4 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/services/WishListService.java (hunk 3 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: don't agree

File: src/main/java/ru/itis/services/WishListService.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: suggestions?

File: src/main/java/ru/itis/services/UserService.java (hunk 6 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/UserService.java]: checks if such user is present in db

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java (hunk 7 above)
Comment (AinaArd) [reply src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: fixed

File: src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java (hunk 8 above)
Comment (AinaArd) [reply src/main/java/ru/itis/security/filters/TokenAuthenticationFilter.java]: removed in properties

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 11 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: If I remove name of the header, it won't be recognized -> 403 error

File: src/main/java/ru/itis/security/SecurityConfig.java (hunk 13 above)
Comment (AinaArd) [reply src/main/java/ru/itis/security/SecurityConfig.java]: not essential

File: src/main/java/ru/itis/dto/ResponseUserDto.java (hunk 10 above)
Comment (AinaArd) [reply src/main/java/ru/itis/dto/ResponseUserDto.java]: fixed

File: src/main/java/ru/itis/services/WishListService.java (hunk 5 above)
Comment (Dimanaux) [reply src/main/java/ru/itis/services/WishListService.java]: - What does `currentUser.getWishLists().add(newWL);` do?
- `return null;` 😢

Comment (Dimanaux) [reply src/main/java/ru/itis/services/WishListService.java]: Why do you query database twice? `userService.findUserByToken(token)`

Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: got it, fixed

File: src/main/java/ru/itis/configuration/Config.java [hunk 14]
```diff
@@ -0,0 +1,14 @@
+package ru.itis.configuration;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class Config {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/configuration/Config.java]: Your bean is duplicated, delete this file.

File: src/main/java/ru/itis/controllers/ProfileController.java [hunk 15]
```diff
@@ -30,29 +30,29 @@ public ProfileController(WishListService wishListService, UserService userServic
     @GetMapping("/profile")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("View user profile page")
-    public UserDto getProfilePage(@RequestHeader(name = "AUTH") String token) {
+    public ResponseUserDto getProfilePage(@RequestHeader(name = "Authorization") String token) {
         Optional<User> userCandidate = userService.findUserByToken(token);
         if (userCandidate.isPresent()) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/ProfileController.java]: You can replace this long if with single `ResponseUserDto.from(defaultUserDto.orElse(defaultUserDto)`

File: src/main/java/ru/itis/controllers/ProfileController.java [hunk 16]
```diff
@@ -30,29 +30,29 @@ public ProfileController(WishListService wishListService, UserService userServic
...
     }
 
     @CrossOrigin
     @DeleteMapping("/profile")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Delete a wish list")
-    public ResponseEntity<?> deleteWishList(@RequestParam String title, @RequestHeader("AUTH") String token) {
+    public ResponseEntity<?> deleteWishList(@RequestParam String title, @RequestHeader("Authorization") String token) {
         if(!wishListService.removeByTitle(title, token)) {
             return new ResponseEntity<>((HttpStatus.NOT_FOUND));
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/ProfileController.java]: Too many braces.

File: src/main/java/ru/itis/controllers/ProfileController.java (hunk 15 above)
Comment (Dimanaux) [reply src/main/java/ru/itis/controllers/ProfileController.java]: Plz read about Optional in Java.

File: src/main/java/ru/itis/services/ItemService.java [hunk 17]
```diff
@@ -21,14 +21,14 @@ public ItemService(ItemsRepository itemsRepository, WishListService wishListServ
...
 
-    public void addNewItem(ItemForm itemForm, Long listId) {
+    public void addNewItem(ItemDto itemDto, Long listId) {
         WishList wishList = getWishList(listId);
         Item newItem = Item.builder()
-                .name(itemForm.getName())
-                .price(itemForm.getPrice())
-                .link(itemForm.getLink())
+                .name(itemDto.getName())
+                .price(itemDto.getPrice())
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/ItemService.java]: Maybe you don't need price.

File: src/main/java/ru/itis/services/UserService.java (hunk 6 above)
Comment (Dimanaux) [reply src/main/java/ru/itis/services/UserService.java]: > checks if such user is present in db

Then it should return a boolean value.

Comment (Dimanaux) [reply src/main/java/ru/itis/services/UserService.java]: Method is too long.

File: src/main/java/ru/itis/services/WishListService.java [hunk 18]
```diff
@@ -37,10 +37,12 @@ public WishList addNewWishList(String title, String token) {
     public boolean removeByTitle(String title, String token) {
         Optional<WishList> wishListCandidate = wishListRepository.findByTitle(title);
         if (wishListCandidate.isPresent()) {
-            User currentUser = userService.findUserByToken(token).get();
-            currentUser.getWishLists().remove(wishListCandidate.get());
-            userService.save(currentUser);
-            wishListRepository.delete(wishListCandidate.get());
+            if (userService.findUserByToken(token).isPresent()) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Too many if statements. Too deep indention.

File: src/main/java/ru/itis/configuration/Config.java (hunk 14 above)
Comment (AinaArd) [reply src/main/java/ru/itis/configuration/Config.java]: fixed

File: src/main/java/ru/itis/services/ItemService.java (hunk 17 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/ItemService.java]: Will need later

File: src/main/java/ru/itis/controllers/ProfileController.java (hunk 15 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/ProfileController.java]: fixed

File: src/main/java/ru/itis/controllers/ProfileController.java (hunk 16 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/ProfileController.java]: fixed

**Pull request merged**